cassovary-core/src/main/scala/com/twitter/cassovary/graph/Node.scala

```diff
@@ -217,12 +221,12 @@ trait Node {
   /**
    * @return Intersection of `dir` neighbors with `nodeIds`.
    */
-  def intersect(dir: GraphDir, nodeIds: Seq[Int]): Seq[Int] = {
+  def intersect(dir: GraphDir, nodeIds: CSeq[Int]): CSeq[Int] = {
     intersect(neighborIds(dir), nodeIds)
   }
 
-  protected def intersect(neighbors: Seq[Int], nodeIds: Seq[Int]): Seq[Int] = {
-    neighbors.intersect(nodeIds)
+  protected def intersect(neighbors: CSeq[Int], nodeIds: CSeq[Int]): CSeq[Int] = {
+    CSeq[Int](neighbors.toSeq.intersect(nodeIds.toSeq).toArray)
```

Comment:
I don't think I understood. Why would we need intersect for Seq[Int] now as it's protected and we want to use CSeq inside cassovary.